service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfProceedingsController.java

```diff
@@ -79,17 +72,16 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody Callback
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
         CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        HearingBooking hearingBooking = hearingBookingService.getMostUrgentHearingBooking(caseData);
-
-        String formattedDate = dateFormatterService.formatLocalDateToString(hearingBooking.getDate() != null
-            ? hearingBooking.getDate() : LocalDate.now(), FormatStyle.LONG);
+        if (caseData.getFamilyManCaseNumber() != null && caseData.getHearingDetails() != null) {
+            HearingBooking hearingBooking = hearingBookingService.getMostUrgentHearingBooking(caseData);
 
-        caseDetails.getData().put("proceedingLabel", String.format("The case management hearing will be on the %s.",
-            formattedDate));
+            caseDetails.getData().put("proceedingLabel", String.format("The case management hearing will be on the %s.",
+                dateFormatterService.formatLocalDateToString(hearingBooking.getDate(), FormatStyle.LONG)));
+        }
 
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseDetails.getData())
-            .errors(eventValidationService.validateEvent(caseData, NoticeOfProceedingsGroup.class))
+            .errors(eventValidationService.validateGroup(caseData, NoticeOfProceedingsGroup.class))
```

Comment:
Couldn't we validate earlier and avoid null checks in the if statement above with the null checks (just check if there are any errors)? It's sort of the purpose of the validation - ensure we can progress with our code.

Reply:
Yes I don't like how we return caseDetails.getData() even if we do have data. Going to split into two returns.